cpp/src/gandiva/cache.h

```diff
@@ -46,12 +51,38 @@ class Cache {
     return result != arrow::util::nullopt ? *result : nullptr;
   }
 
+  ValueType GetObjectCode(KeyType cache_key) {
+    arrow::util::optional<ValueType> result;
+    mtx_.lock();
+    result = cache_.getObject(cache_key);
+    mtx_.unlock();
+    if (result != arrow::util::nullopt) {
+      return *result;
+    } else {
+      return nullptr;
+    }
+  }
+
   void PutModule(KeyType cache_key, ValueType module) {
     mtx_.lock();
     cache_.insert(cache_key, module);
     mtx_.unlock();
   }
 
+  void PutObjectCode(KeyType& cache_key, ValueType object_code, size_t object_cache_size) {
+    mtx_.lock();
+    cache_.insertObject(cache_key, object_code, object_cache_size);
+    mtx_.unlock();
+  }
+
+  std::string toString() {
+    return cache_.toString();
+  }
+
+  size_t getCacheSize(){
+    return cache_.getLruCacheSize();
+  }
```

Comment:
By default the methods names starts with capital letter, these functions must follow that pattern right?

Reply:
Done